services/transactionpool/init.go

```diff
@@ -71,6 +74,14 @@ func NewTransactionPool(ctx context.Context,
 	return s
 }
 
+func createClockIfNeeded(maybeClock adapter.Clock) adapter.Clock {
+	if maybeClock == nil {
+		return adapter.NewSystemClock()
```

Comment:
wouldn't it be more readable to use `adapter.NewSystemClock()` instead of `nil` in all the places you pass nil to maybeClock? this would remove the use of the notation for "optional" which is called maybe. this "maybe" convention is not very clear to me, why not optional - or better yet - just make it mandatory.  it would also remove the need for the cryptic "if needed" expression here..

Reply:
Maybe; the tradeoff here is that I need `AdjustableClock` to be shared by all nodes in a Gamma process, while `systemClock` can (and most likely _must_) be independently instantiated for each node, so I would need to find a way to do both. This is kinda like `nodeDependencyProvider` in `inmemory.Network`, where Gamma passes it as empty, indicating that `inmemory.NewNetworkWithNumOfNodes` should create the dependencies on its own. There might be a refactoring hiding here somewhere, I'm not sure yet.

Reply:
I see your point, it's valid in principle but there is also an argument for allowing `systemClock` to be shared, assuming it's a shallow wrapper around `time.Now()` by definition.. 

A clockBuilder would solve this issue somewhat verbosely (granted it's ugly but if you make it a generator function maybe the overhead can be minimized).